### PR TITLE
feat: include more information in system.operations table

### PR DESCRIPTION
**Pull request by tustvold**
This PR does a number of things in the interests of improving the system.operations table

* It incorporates the task tracker error reporting added in #2031  to report success/failure
* It changes the chunk_id to instead be a comma-separated string of chunk_id**s** (arrow does have structured types but I had no idea what would happen if I used them)
* It includes the table name of the job if any

It also contains two hopefully uncontroversial cleanups:

* Changes `Job` to use `ChunkAddr` and `PartitionAddr`
* Renames `Job::CloseChunk` to `Job::CompactChunk`

### CODE REVIEW

File: generated_types/src/job.rs [hunk 1]
```diff
@@ -1,70 +1,54 @@
...
+                Self::CompactChunks(management::CompactChunks {
+                    db_name: partition.db_name.to_string(),
+                    partition_key: partition.partition_key.to_string(),
+                    table_name: partition.table_name.to_string(),
+                    chunks,
+                })
+            }
+            Job::PersistChunks { partition, chunks } => {
+                Self::PersistChunks(management::PersistChunks {
+                    db_name: partition.db_name.to_string(),
```
Comment (tustvold) [inline generated_types/src/job.rs:41]: Perhaps we could introduce PartitionAddr and ChunkAddr into the gRPC API and eliminate some of this boilerplate - but something for another time

File: server/src/lib.rs [hunk 2]
```diff
@@ -1827,19 +1828,24 @@ mod tests {
...
-        let db_name_string = db_name.to_string();
+        let chunk_addr = ChunkAddr {
+            db_name: Arc::from(db_name.as_str()),
+            table_name: Arc::from("cpu"),
+            partition_key: Arc::from(""),
+            chunk_id: 0,
+        };
         let tracker = server
-            .close_chunk(db_name, table_name, partition_key, 0)
+            .close_chunk(
```
Comment (tustvold) [inline server/src/lib.rs:1838]: Hopefully we can move more of the API to using ChunkAddr and PartitionAddr and cleanup some of this

File: data_types/src/job.rs [hunk 3]
```diff
@@ -11,87 +16,87 @@ pub enum Job {
...
-        chunk_id: u32,
+        chunk: ChunkAddr,
     },
 
     /// Compact a set of chunks
     CompactChunks {
-        db_name: String,
-        partition_key: String,
-        table_name: String,
+        partition: PartitionAddr,
```
Comment (alamb) [inline data_types/src/job.rs:30]: 👍  nice cleanups

File: generated_types/src/job.rs (hunk 1 above)
Comment (alamb) [reply generated_types/src/job.rs:41]: Having them be separate fields on gRPC may make writing clients easier 

Maybe we can remove the boiler plate with a macro or something

File: query_tests/src/sql.rs [hunk 4]
```diff
@@ -347,20 +347,20 @@ async fn sql_select_from_system_chunk_columns() {
...
-        "| 0  | Complete | true          | true           | 1970-01-01T00 | 0        | Loading chunk to ReadBuffer         |",
-        "| 1  | Complete | true          | true           | 1970-01-01T00 |          | Persisting chunks to object storage |",
-        "| 2  | Complete | true          | true           | 1970-01-01T00 | 2        | Writing chunk to Object Storage     |",
-        "+----+----------+---------------+----------------+---------------+----------+-------------------------------------+",
+        "+----+---------+---------------+----------------+------------+---------------+-----------+-------------------------------------+",
+        "| id | status  | took_cpu_time | took_wall_time | table_name | partition_key | chunk_ids | description                         |",
+        "+----+---------+---------------+----------------+------------+---------------+-----------+-------------------------------------+",
+        "| 0  | Success | true          | true           | h2o        | 1970-01-01T00 | 0         | Compacting chunk to ReadBuffer      |",
```
Comment (alamb) [inline query_tests/src/sql.rs:353]: 👍

File: server/src/db/lifecycle/move_chunk.rs [hunk 5]
```diff
@@ -24,11 +24,8 @@ pub fn move_chunk_to_read_buffer(
     let db = Arc::clone(&guard.data().db);
     let addr = guard.addr().clone();
     // TODO: Use ChunkAddr within Job
```
Comment (alamb) [inline server/src/db/lifecycle/move_chunk.rs:26]:
```suggestion
```

File: server/src/db/lifecycle/write.rs [hunk 6]
```diff
@@ -56,10 +56,7 @@ pub(super) fn write_chunk_to_object_store(
 
     // TODO: Use ChunkAddr within Job
```
Comment (alamb) [inline server/src/db/lifecycle/write.rs:57]:
```suggestion
```

**Review by alamb — APPROVED** [review]
This looks like a really nice change to me

> It changes the chunk_id to instead be a comma-separated string of chunk_ids (arrow does have structured types but I had no idea what would happen if I used them)

I think this is a wise move. I don't think `Struct` arrays can be pretty printed yet, though https://github.com/apache/arrow-rs/pull/579 from @houqp is fixing that :)

**Pull request merged**